client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java

```diff
@@ -156,6 +157,54 @@ public void testGetWithType() {
         getAndExistsWithTypeTest(RequestConverters::get, HttpGet.METHOD_NAME);
     }
 
+    public void testSourceExists() throws IOException {
+        String index = randomAlphaOfLengthBetween(3, 10);
+        String id = randomAlphaOfLengthBetween(3, 10);
+        final GetRequest getRequest;
+        if (randomBoolean()) {
+            getRequest = new GetRequest(index, id);
```

Comment:
Instead of using randomization, we've been creating two separate tests to make sure that we always test both typed + typeless requests (see `testExplain`/ `testExplainWithTypes` for example). This is something we settled on recently, my reference PR was probably out-of-date and led you down this path.

Reply:
Actually it was a consciuos decision because I don't think the double testing is really necessary, we run so many tests in CI that this will trigger both variants quickly. But I will add a second test if that is what you decided.

Reply:
Got it. I agree we should go with the two tests now for consistency.